src/components/clerkcustom/invite-form.tsx

```diff
@@ -0,0 +1,207 @@
+'use client'
+
+import { TextInput, PasswordInput, Button, Alert, Checkbox, Group, Text } from '@mantine/core'
+import React, { useState } from 'react'
+import { createClerkUserAction } from '@/server/actions/clerk-user-actions'
+import { Warning } from '@phosphor-icons/react'
+import { OrganizationSelect } from '@/components/clerkcustom/organization-select'
+import Link from 'next/link'
+import { createUserAction } from '@/server/actions/user-actions'
+
+export default function InviteForm() {
+    const [error, setError] = useState<string | null>(null)
+    const [loading, setLoading] = useState(false)
+    const [success, setSuccess] = useState(false)
+    const [formValues, setFormValues] = useState({
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        roles: {
+            reviewer: false,
+            researcher: false,
+        },
+    })
+    const [selectedOrganization, setSelectedOrganization] = useState<string | null>(null)
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault()
+        setError(null)
+        setSuccess(false)
+        setLoading(true)
+
+        const formData = new FormData(e.currentTarget)
+        const firstName = formData.get('firstName')?.toString() ?? ''
+        const lastName = formData.get('lastName')?.toString() ?? ''
+        const email = formData.get('email')?.toString() ?? ''
+        const password = formData.get('password')?.toString() ?? ''
+
+        // Check if organization is selected
+        if (!selectedOrganization) {
+            setError('Please select an organization')
+            setLoading(false)
+            return
+        }
+
+        // Check if at least one role is selected
+        if (!formValues.roles.reviewer && !formValues.roles.researcher) {
+            setError('Please select at least one role (Reviewer or Researcher)')
+            setLoading(false)
+            return
+        }
+
+        // Update form values state
+        setFormValues({
+            firstName,
+            lastName,
+            email,
+            password,
+            roles: formValues.roles, // Keep the existing roles structure
+        })
+
+        try {
+            // save user in Clerk
+            const result = await createClerkUserAction({
+                firstName,
+                lastName,
+                email,
+                password,
+            })
+            // save user in DB
+            await createUserAction(
+                result.clerkId,
+                `${firstName} ${lastName}`,
+                formValues.roles.researcher,
+                selectedOrganization, // Use the selected organization ID directly
+                formValues.roles.reviewer,
+            )
```

Comment:
something to keep in mind is that all actions happen over the network the same as if you were firing off `fetch` calls.  This means that you're relying on the user having a stable network connection and not closing the page, navigating away etc so that they both occur.  it'd be much better to create a single action that performs these steps inside a transaction if possible